src/test/java/com/stripe/BaseStripeTest.java

```diff
@@ -506,18 +508,77 @@ public ParamMapMatcher(Map<String, Object> other) {
      * Informs if this matcher accepts the given argument.
      */
     @Override
-    public boolean matches(Map<String,Object> paramMap) {
+    public boolean matches(Map<String, Object> paramMap) {
       if (this.other == null) {
         // If the matcher was constructed with null, accept any params
         return true;
       } else {
         if (paramMap == null) {
           return this.other.isEmpty();
         } else {
-          return this.other.equals(paramMap);
+          return compareParamObjects(this.other, paramMap);
         }
       }
     }
+
+    @Override
+    public String toString() {
+      return other.toString();
+    }
+  }
+
+  private static Boolean compareParamObjects(Object thisValue, Object otherValue) {
```

Comment:
Additional logic to make testing of typed param and comparison with old untyped map easier.
In the subsequent PR, I rely on the existing tests that has untyped params. I want to make sure that using typed params will have the same effect.